tests/test_schemas.py

```diff
@@ -1,294 +1,645 @@
+        elif template_type == 'class':
+            template = data_model_jsonld.class_template()
+
+        try:
+            # Fill out template for given node.
+            object_template = data_model_jsonld.create_object(template=template, node=node)
+            # Ensure template keys are present (not all original keys will be present due to cleaning empty values):
+        except:
+            # Should only fail if no node is given
+            assert node == ''
+
+        if 'object_template' in locals():
+            # Check that object template keys match the expected keys
+            actual_keys = list(object_template.keys())
+            if template_type == 'property':
+                expected_keys = ['@id', '@type', 'rdfs:comment', 'rdfs:label', 'schema:isPartOf', 'sms:displayName', 'sms:required', 'sms:validationRules']
+            elif template_type == 'class':
+                expected_keys = ['@id', '@type', 'rdfs:comment', 'rdfs:label', 'rdfs:subClassOf', 'schema:isPartOf', 'sms:displayName', 'sms:required', 'sms:requiresDependency', 'sms:validationRules']
+            assert (set(actual_keys) - set(expected_keys)) == (set(expected_keys) - set(actual_keys))
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    @pytest.mark.parametrize("template_type", ['property', 'class'], ids=['property', 'class'])
+    def test_add_contexts_to_entries(self, helpers, data_model, template_type):
+        # Will likely need to change when contexts added to model.
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Get empty template
+        if template_type == 'property':
+            template = data_model_jsonld.property_template()
+        elif template_type == 'class':
+            template = data_model_jsonld.class_template()
+
+        # Fill out template for given node.
+        object_template = data_model_jsonld.create_object(template=template, node='Patient')
+
+        if 'sms:required' in object_template:
+            assert 'sms' in object_template['sms:required']
+        if '@id' in object_template:
+            assert 'bts' in object_template['@id']
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_clean_template(self, helpers, data_model):
+        # TODO: This will need to change with contexts bc they are hard coded here.
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Get empty template
+        template = data_model_jsonld.class_template()
+        assert 'sms:requiresDependency' in template
```

Comment:
Nit: When I was first reading the test, i was a bit confused why here only `sms:requiresDependency` got tested in assert, but it turned out that other keys such as `sms:requiresComponent` and `schema:rangeIncludes` got deleted as well. Maybe it will be worth to clarify this in a comment. 

But in general, i like how this test was written. I think it captures everything that needs to be tested in the `clean_template` function 

--Migrated from https://github.com/Sage-Bionetworks/schematic/pull/1313